tests/test_plugins/test_design.py

```diff
@@ -175,113 +344,330 @@ def scs(radius: float, num_spheres: int, tag: str) -> float:
+            run_time=1e-12,
+            monitors=[mnt],
+        )
 
-    # make sure returning a list assigns column labels properly
+    def scs_post_aux(sim_data):
+        """Uses scs_pre for pre function"""
+        mnt_data = sim_data["field"]
+        ex_values = mnt_data.Ex.values
 
-    num_outputs = 3
-    label_keys = tdd.Result.default_value_keys(num_outputs * [0.0])
+        return (
+            np.sum(np.square(np.abs(ex_values))),
+            {"test1": True, "test2": None, "test3": "a great success", "test4": 3.14},
+        )
 
-    def scs_list(*args, **kwargs):
-        output = scs(*args, **kwargs)
-        return num_outputs * [output]
+    # Setup Designspace
+    radius_variable = tdd.ParameterFloat(
+        name="radius",
+        span=(0, 1.5),
+        num_points=5,  # note: only used for MethodGrid
+    )
 
-    df = design_space.run(scs_list).to_dataframe()
-    for label in label_keys:
-        assert label in df, "dict key not parsed properly as column header"
-        for value in df[label]:
-            assert not isinstance(value, (tuple, list)), "dict saved instead of value"
+    num_spheres_variable = tdd.ParameterInt(
+        name="num_spheres",
+        span=(0, 3),
+    )
+
+    tag_variable = tdd.ParameterAny(name="tag", allowed_values=("tag1", "tag2", "tag3"))
+
+    design_space = tdd.DesignSpace(
+        parameters=[radius_variable, num_spheres_variable, tag_variable],
+        method=sweep_method,
+        name="sphere CS",
+        task_name=f"{sweep_method.type}",
+    )
+
+    # Include auxiliary data in list format in the output for non_td run
+    aux_not_td_sweep = design_space.run(float_non_td_pre, float_non_td_aux_post)
+    aux_not_td_df = aux_not_td_sweep.to_dataframe(include_aux=True)
+
+    assert aux_not_td_df["aux_key_2"][0] == "data"
+
+    # Test for auxiliary values as dict format in output of td function
+    ts_sim_aux = design_space.run(scs_pre, scs_post_aux)
+    ts_sim_aux_df = ts_sim_aux.to_dataframe(include_aux=True)
+
+    assert ts_sim_aux_df["test4"][0] == 3.14
+
+    # Test that sampler style complex input fails properly
+    with pytest.raises(ValueError):
+        ts_sim_complex = design_space.run(scs_pre, scs_post_complex_return)
+
+    # Test that wrongly formatted aux outputs throw an error
+    def bad_format_non_td_aux_post(res):
+        """Uses the same float_non_td_pre as pre"""
+        return ["not a float", int(res), ["any", "other", "data"]]
+
+    with pytest.raises(ValueError):
+        bad_format_td_aux_sweep = design_space.run(float_non_td_pre, bad_format_non_td_aux_post)
+
+    # Test that plots have been produced and stored
+    assert ts_sim_aux.opt_output is not None
+
+    # Create an import error to test optimizer can error if relevant package is not installed
+    # Placed at the end of the test so that module is loaded for other checks
+
+    with pytest.raises(ImportError):
+        sys.modules[method_module_convert[sweep_method.type]] = None
```

Comment:
Setattr doesn't seem to do it because I need the whole module to ImportError later on. I've changed `sys.modules` with `monkeypatch.setitem` which I might be the best compromise.